core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt

```diff
@@ -165,47 +153,59 @@ class MotorController<T : MotorId>(
                 field = value
             }
 
-    fun setPercentOutput(output: Double) {
+    val percentOutput: Double
+        get() = ctreMotorController.motorOutputPercent
+
+    var position: Int
+        get() = ctreMotorController.getSelectedSensorPosition(0)
+        set(value) {
+            ctreMotorController.selectedSensorPosition = value
+        }
+
+    fun position(unit: MetricUnit<Angular>) =
+            MetricValue(encoder!!.ticks, position.toDouble()).convertTo(unit)
+
+    val velocity: Int
+        get() = ctreMotorController.getSelectedSensorVelocity(0)
+
+    fun velocity(unit: CompositeUnit<Per, Angular, Chronic>) =
+            MetricValue(encoder!!.ticksPerSecond, velocity.toDouble()).convertTo(unit)
+
+    fun targetPercentOutput(output: Double) {
         ctreMotorController.set(CtreControlMode.PercentOutput, output)
     }
 
-    fun setPosition(position: Int) {
+    fun targetPosition(position: Int) {
         ctreMotorController.set(CtreControlMode.Position, position.toDouble())
     }
 
-    fun <U : MetricUnit<Angular>> setPosition(position: MetricValue<Angular, U>) {
+    fun <U : MetricUnit<Angular>> targetPosition(position: MetricValue<Angular, U>) {
         try {
-            setPosition(position.convertTo(encoder!!.ticks).value.toInt())
+            targetPosition(position.convertTo(encoder!!.ticks).value.toInt())
         } catch (e: NullPointerException) {
             throw java.lang.IllegalStateException(
                     "You must configure your encoder to use units."
             )
         }
     }
 
-    fun position(unit: MetricUnit<Angular>) =
-        MetricValue(encoder!!.ticks, sensorPosition.toDouble()).convertTo(unit)
-
-    fun setVelocity(velocity: Double) {
-        ctreMotorController.set(CtreControlMode.Velocity, velocity)
+    fun targetVelocity(velocity: Int) {
+        ctreMotorController.set(CtreControlMode.Velocity, velocity.toDouble())
     }
 
-    fun setVelocity(
+    fun targetVelocity(
         velocity: MetricValue<CompositeUnitType<Per, Angular, Chronic>, CompositeUnit<Per, Angular, Chronic>>
     ) {
         try {
-            setVelocity(velocity.convertTo(encoder!!.ticksPerSecond).value)
+            targetVelocity(velocity.convertTo(encoder!!.ticksPerSecond).value.toInt())
         } catch (e: NullPointerException) {
             throw java.lang.IllegalStateException(
                     "You must configure your encoder to use units."
             )
         }
     }
 
-    fun velocity(unit: CompositeUnit<Per, Angular, Chronic>) =
-        MetricValue(encoder!!.ticksPerSecond, sensorVelocity.toDouble()).convertTo(unit)
-
-    fun setCurrent(current: Double) {
+    fun targetCurrent(current: Double) {
```

Comment:
```suggestion
    fun setCurrent(current: Double) {
```
This change also takes effect almost immediately.

Reply:
So now `targetPosition` is `setTargetPosition` and `targetPercentOutput` is just `setPercentOutput`.